### PR TITLE
fix: proxy events into instance var and add test script

### PR DESCRIPTION
**Pull request by akoserwal**
## Pre-Submission Checklist

- [ x] Opened an issue discussing these changes before opening the PR
- [ x] Ran the linter and tests via `make prepush`
- [ x] Included comprehensive and convincing tests for changes

## Issues
#1348 
Closes:

* Issue #1348
* Issue #
* Issue #

> Summarize the issues that discussed these changes
Since PROXY_EVENTS are shared between every instance of a restify server, and upgrade is added but never removed, all future instances of Server will have the upgrade event proxied.

# Changes

> What does this PR do?
update PROXY_EVENTS to an instance variable.

### CODE REVIEW

**Comment by akoserwal** [comment]
@DonutEspresso, @hekike : I created a new branch. Instead of rebase my previous branch. I accidentally clicked over update branch on the github uI. Which will add non-essential commits. I will take care from next time.

**Comment by DonutEspresso** [comment]
Thanks @akoserwal! Apologies for the trouble. I left some minor comments - let me know if they make sense!

File: test/server.test.js [hunk 1]
```diff
@@ -2430,3 +2430,19 @@ test('should emit error with multiple next calls with strictNext', function(t) {
         });
     });
 });
+
+test('should have proxy event handlers as instance', function(t) {
+    var server = restify.createServer({
+        handleUpgrades: false
+    });
+    t.equal(server.proxyEvents.length, 7);
```
Comment (DonutEspresso) [inline test/server.test.js]: I think I expect handleUpgrades: true to be 7, and 6 for false (may be related to the && indexOf check above)

File: lib/server.js [hunk 2]
```diff
@@ -187,10 +187,10 @@ function Server(options) {
 
     this.router.on('mount', this.emit.bind(this, 'mount'));
 
-    if (!options.handleUpgrades && PROXY_EVENTS.indexOf('upgrade') === -1) {
-        PROXY_EVENTS.push('upgrade');
+    if (!options.handleUpgrades && this.proxyEvents.indexOf('upgrade') === -1) {
```
Comment (DonutEspresso) [inline lib/server.js]: Now that proxyEvents is specific to the instance of the server being created, I don't think we even need the indexOf check right?

File: test/server.test.js (hunk 1 above)
Comment (akoserwal) [reply test/server.test.js]: As per the doc

options.handleUpgrades Boolean Hook the upgrade event from the node HTTP server, pushing Connection: Upgrade requests through the regular request handling chain. (optional, default false) (http://restify.com/docs/server-api/). 

We are updating proxyEvents array when 
`options.handleUpgrades:false`
size will be 7 and in-case of true. we will skip it.

File: lib/server.js (hunk 2 above)
Comment (akoserwal) [reply lib/server.js]: @DonutEspresso: right, I can remove it.

File: test/server.test.js (hunk 1 above)
Comment (DonutEspresso) [reply test/server.test.js]: Ah, I can't read. Must have been a long day. :D

**Comment by DonutEspresso** [comment]
@akoserwal thank you for the contribution! 🎉

**Comment by akoserwal** [comment]
@DonutEspresso : Thank you :)

**Pull request merged**